lib/resty/upstream/healthcheck.lua

```diff
@@ -201,8 +199,14 @@ local function peer_ok(ctx, is_backup, id, peer)
     end
 end
 
+-- shortcut error function for check_peer()
+local function report_error(sock, ctx, is_backup, id, peer, ...)
+  if not peer.down then errlog(...) end
+  sock:close()
```

Comment:
removed the extra close call, see https://github.com/Tieske/lua-resty-upstream-healthcheck/commit/dfc13d5e263ce01f75e3393d84242b68f6571500, but did so by duplicating 1 errorlog call, instead of duplicating 4 close calls. Performance same, but code is tighter this way.